test/jdk/java/awt/print/PrinterJob/JobName/PrinterJobName.java

```diff
@@ -21,173 +21,59 @@
  * questions.
  */
 
-/**
+import java.awt.Graphics;
+import java.awt.Graphics2D;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterJob;
+
+/*
  * @test
  * @bug 4205601
  * @summary setJobName should be used by PrinterJob
  * @key printer
+ * @library /java/awt/regtesthelpers
+ * @library /test/lib
+ * @build PassFailJFrame
+ * @build jtreg.SkippedException
  * @run main/manual PrinterJobName
  */
-
-import java.awt.*;
-import java.awt.print.*;
-
 public class PrinterJobName implements Printable {
 
+    static String theName = "Testing the Jobname setting";
 
-  static String theName = "Testing the Jobname setting";
-
-  public static void main(String[] args) {
+    private static final String INSTRUCTIONS =
+            "You must have a printer available to perform this test\n" +
+            "This test prints a page with a banner/job name of\n" + theName;
 
-       String[] instructions =
-        {
-         "You must have a printer available to perform this test",
-         "This test prints a page with a banner/job name of",
-          theName
-       };
+    public static void main(String[] args) throws Exception {
 
-      Sysout.createDialog( );
-      Sysout.printInstructions( instructions );
+        if (PrinterJob.lookupPrintServices().length == 0) {
+            throw new SkippedException("Printer not configured or available."
+                    + " Test cannot continue.");
+        }
 
-      PrinterJob job = PrinterJob.getPrinterJob();
-      job.setJobName(theName);
-      job.setPrintable(new PrinterJobName());
-      try {
-          job.print();
-          System.out.println("PRINTING DONE.");
-      }
-      catch (Exception exc) {
-          System.out.println("Printer Exception");
-      }
-  }
+        PassFailJFrame passFailJFrame = new PassFailJFrame.Builder()
+                .instructions(INSTRUCTIONS)
+                .rows((int) INSTRUCTIONS.lines().count() + 1)
+                .columns(45)
+                .build();
 
+        PrinterJob job = PrinterJob.getPrinterJob();
```

Comment:
this can be moved into helper function

Reply:
Not really… unless you create secondary UI to initiate printing.